op-node/metrics/metrics.go

```diff
@@ -516,8 +530,10 @@ func (m *Metrics) RecordDerivedBatches(batchType string) {
 	m.DerivedBatches.Record(batchType)
 }
 
-func (m *Metrics) CountSequencedTxs(count int) {
-	m.TransactionsSequencedTotal.Add(float64(count))
+func (m *Metrics) CountSequencedTxsInBlock(txns int, deposits int) {
+	m.TransactionsSequencedTotal.WithLabelValues("l1_info_deposit").Add(1)
+	m.TransactionsSequencedTotal.WithLabelValues("deposits").Add(float64(deposits - 1))
```

Comment:
Interop is adding another system tx that backruns the L1 info deposits tx. I would prefer to just count all deposits including L1 attributes, so this doesn't ever get out of sync. We don't want hardfork branching logic in here. It also assumes that there is always a deposit tx, which is true and unlikely to change, but this is fragile to that app logic detail given the subtraction

Reply:
Fair that this is annoying to do across HF. I do like the idea of splitting out system transactions from user transactions, but even the difference from deposits to normal L2 transactions provides useful data.